### DISCUSSION TITLE
Probleme de pas du moteur

### DISCUSSION THREAD

**Discussion by Dididu12** [kachurovskiy/nanoels#40] [Q&A] [answered]
![16418291563371985171199517232758](https://user-images.githubusercontent.com/97460665/148793825-f2253a1b-136a-4994-91ba-d923d55ceb62.jpg)
![16418291998261176866691212269984](https://user-images.githubusercontent.com/97460665/148793890-bc3964b4-aed6-455c-8e0d-34832ead9e94.jpg)
![Uploading 16418292176317595744863086753815.jpg…]()

**Comment by kachurovskiy** [comment, marked as answer]
Dilan, looks like your 3rd picture wasn't fully uploaded, please re-attach it.

So from what I understand, your problem is that you expected the cross-slide to move 2.53mm but it only moved 1mm? You need to get the following data from your hardware:

1. Number of pulses per revolution from your encoder
2. Number of steps your motor takes per revolution (usually 200, 400, 800, etc. - set on the driver)
3. Lead screw pitch - could it be that you have an imperial lead screw with a 2.54 pitch?

Once you have these numbers, fill them in https://github.com/kachurovskiy/nanoels/blob/main/h1/NanoEls.ino#L6-L8 and re-upload the sketch to your Arduino.

Cheers!

**Reply by Dididu12** [reply]
Merci je vais essayer sa J'ai un autre question pour les bouton gauche droite sont inversés comment résoudre se problème merci

**Reply by kachurovskiy** [reply]
Change line 24 to a different value from what it is now (`false` or `true`) - https://github.com/kachurovskiy/nanoels/blob/main/h1/NanoEls.ino#L24

**Reply by Dididu12** [reply]
Bonjour J'ai deux questions 

1:C'est possible de mettre le programme a deux AXE c'est quoi qui ne va pas avec ce code c'est plutôt cool je trouve

2:Les problème ont l'air résolu mais il n'y a rien dans le code pour lui indiquer le jeu de la noix sur la vis mère?? merci

**Reply by kachurovskiy** [reply]
1. NanoEls only supports 1 axis, for 2-axis ELS you have https://www.rocketronics.de/shop/en/ or https://github.com/kachurovskiy/megaels

2. NanoEls doesn't account for the backlash, it's on the operator - just like when you're operating with gears.

**Reply by Dididu12** [reply]
D'autre question 

La dernière vidéo sur Youtube ma plus avec les joystick j'espère que se nouveau projet va être fonctionnel aller vous le mettre sur git hub??

J'ai acheté un driver de marque rtelligent R60 pour les réglage il et possible de mettre 200 step
Je le règle la dessus ??

Mon moteur et un 5.0A je peut régler sur 4.9A avec mon driver??

Merci une fois fini le tour je posterai en exemple de build si possible merci maxim

**Reply by kachurovskiy** [reply]
Not sure if I understand your first question fully.

For R60, according to the manual http://www.rtelligent.net/upload/wenjian/R60_User_Manual.pdf you can set 200 steps by setting SW5, SW6, SW7 and SW8 to "ON". That said the motor might be too noisy or rough with 200 steps, you might get better results with 400 or 800 steps.

Setting high current on the open-loop driver will result in excessive heat generation on both motor and driver, I would set the lowest current sufficient for your operations, maybe start with ~2A?

Please do post an overview of your build once finished, many thanks!

**Reply by Dididu12** [reply]
Ma première question et 

Aller vous rendre cette
version exploitable et fonctionnel pour tous merci 
https://github.com/kachurovskiy/megaels

**Reply by kachurovskiy** [reply]
Dilan, since you've got this far with NanoEls I would recommend finishing this build first.

MegaEls is based on a mature codebase and I've got it working reasonably well on my lathe right now but it has a steep learning curve. I might also change the PCB and case in the coming months a bit. I would suggest waiting half a year before starting building it.

You might also want to consider Rocketronics as it's the best and easiest to use option today - if somewhat pricey.

**Reply by Dididu12** [reply]
![20220122_104056](https://user-images.githubusercontent.com/97460665/150633904-7f74a93e-66c3-423f-a2c0-74d5b8a0db51.jpg)
![20220110_143043_remastered](https://user-images.githubusercontent.com/97460665/150633927-6bad61e4-b7af-4abd-b497-b8f75d0c4e04.jpg)
Problème résolu le moteur prend bien ses pas comme il faut maintenant c'est un projet officielement abouti je remercie encore kachurovskiy
![20220111_062541](https://user-images.githubusercontent.com/97460665/150610674-e3c70aa5-b28a-42aa-9f8f-f590c650a6a9.jpg)
![20220111_062547](https://user-images.githubusercontent.com/97460665/150610713-61fe01cc-278e-4391-b154-0130400f0532.jpg)
![20220111_062557](https://user-images.githubusercontent.com/97460665/150610729-c7d4839a-84e5-4d82-91cc-ac1b649056f0.jpg)
![20220122_104110](https://user-images.githubusercontent.com/97460665/150633708-36e33dba-a9eb-4b99-8dff-ef9b6cf0b58f.jpg)
![20220122_104153](https://user-images.githubusercontent.com/97460665/150633726-9ea89370-bd6f-4986-bbbb-4074b6dbda2b.jpg)
![20220122_104659](https://user-images.githubusercontent.com/97460665/150633735-2b373c22-c650-4a46-9f90-3ba0ec1b736e.jpg)
![20220122_104805](https://user-images.githubusercontent.com/97460665/150633746-e5f65e15-4cb5-49a1-973f-7e8ad5fa5ab3.jpg)
![20220122_104901](https://user-images.githubusercontent.com/97460665/150633758-e43e6f1b-edfc-4c9e-9c58-4cb9b9355ed5.jpg)

**Reply by kachurovskiy** [reply]
Looks great, well done! 🥳🎉

**Reply by Dididu12** [reply]
https://youtu.be/6ghRS3S5NTI

**Reply by Dididu12** [reply]
Je vous enverrai d'autres photo pour mettre sur le github si possible en exemple de build merci

**Reply by Dididu12** [reply]
[Marquage Boite (2).docx](https://github.com/kachurovskiy/nanoels/files/7918069/Marquage.Boite.2.docx)
C'est le marquage de la boîte un document word a modifier avec vos préférences c'est moi qui l'ai réalisé 

Il faut l'imprimer a l'échelle 1:1 si problème me contacter a la suite de se message merci
